Utilities/XrdAdaptor/README.md

```diff
@@ -0,0 +1,49 @@
+# XrdAdaptor
+
+## Introduction
+
+The `XrdAdaptor` package is the CMSSW implementation of CMS' AAA infrastructure. The main features on top of the stock XRootD client library are
+* Recovery from some errors via re-tries
+* Use of multiple XRootD sources (described further [here](doc/multisource_algorithm_design.md))
+
+## Short description of components
+
+### `ClientRequest`
+
+The `ClientRequest` implements `XrdCl::ResponseHandler`, and represents a single read request(?).
+
+### `QualityMetric` etc
+
+?
+
+### `RequestManager`
+
+The `RequestManager` containes the actual implementation of the retries and the multi-source algorithm. There is one `RequestManager` object for one PFN, and it contains one or more `Source` objects.
+
+#### `RequestManager::OpenHandler`
+
+The `OpenHandler` implements XRootD's `XrdCl::ResponseHandler` in an asynchronous way(?). An instance is created in `RequestManager::initialize()`, and used when additional Sources are opened, either as part of the multi-source comparisons (`RequestManager::checkSourcesImpl()`) or read error recovery (`RequestManager::requestFailure()`).
```

Comment:
Correct -- the `OpenHandler` is async.

Most of the internal operations for the xrootd client are asynchronous while ROOT expects a synchronous interface from the adaptor.  The difference between the two here is the asynchronous one is used for "background probing" to find additional sources.